File: sorting/bubblesort.cpp
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void bubblesort(int* array, int length)
{
    int i, j;
    for (i = 0; i < length - 1; ++i) {
       for (j = 0; j < length - i - 1; ++j) {
           if (array[j] > array[j + 1]) {
               int tmp = array[j];
               array[j] = array[j + 1];
               array[j + 1] = tmp;
           }
       }
    }
}

int main(int argc, const char** argv)
{
    FILE *f = fopen(argv[1], "r");
    int size, i, j;
    int *array, *array2;
    i = 0;
    while (i == 0) {
        i = fread(&size, sizeof(int), 1, f);
        printf("i: %d\n", i);
    }
    printf("size: %d\n", size);
    array = (int*)malloc(size * sizeof(int));
    i = size;
    while (i > 0) {
        j = fread(array, sizeof(int), i, f);
        i -= j;
    }

    //array2 = (int*)malloc(size * sizeof(int));
    printf("Sorting %d data points...\n", size);
    //for (i=0; i < 5; ++i) {
        //memcpy(array2, array, size * sizeof(int));
        //bubblesort(array2, size);
        bubblesort(array, size);
    //}

    //for (i=0; i < size; ++i) {
    //    printf("%d\n", array[i]);
    //}

    return 0;
}

File: sorting/BubbleSort.java
import java.io.*;

public class BubbleSort { 
    public static void sort(int[] x) {
         for (int i=0; i < x.length - 1; ++i) {
             for (int j=0; j < x.length - i - 1; ++j) {
                 if (x[j] > x[i + 1]) {
                     int tmp = x[j];
                     x[j] = x[j + 1];
                     x[j + 1] = tmp;
                 }
             }
         }
    }
    
    public static void main(String[] args) throws FileNotFoundException, IOException {
        DataInputStream in = new DataInputStream(
            new BufferedInputStream(
                new FileInputStream(args[0])
            )
        );
        int size = in.readInt();
        int[] array = new int[size];
        for (int i=0; i < size; ++i) {
            array[i] = in.readInt();
        }

        //int[] array2 = new int[array.length];
        System.out.println("Sorting " + size + " data points...");
        //for (int i=0; i < 5; ++i) {
        //    System.arraycopy(array, 0, array2, 0, array.length);
        //    sort(array2);
            sort(array);
        //}
        //for (int i=0; i < array.length; i++) 
        //    System.out.println(array[i]);    
        //} 
    }
}

